pallets/order-book/src/lib.rs

```diff
@@ -466,11 +469,19 @@ pub mod pallet {
                 Error::<T>::NotAllowedDEXId
             );
             let dex_info = T::DexInfoProvider::get_dex_info(&order_book_id.dex_id)?;
-            // the base asset of DEX must be a quote asset of order book
+            // a quote asset of order book must be the base asset of DEX
             ensure!(
                 order_book_id.quote == dex_info.base_asset_id,
-                Error::<T>::NotAllowedBaseAsset
+                Error::<T>::NotAllowedQuoteAsset
             );
+
+            // synthetic asset are forbidden
+            ensure!(
+                order_book_id.base != dex_info.synthetic_base_asset_id
```

Comment:
Do we really want to forbid XOR-XST pair in Order book?

Reply:
Fixed